src/raftstore/store/peer.rs

```diff
@@ -1349,20 +1349,28 @@ impl Peer {
                 // have no effect.
                 self.proposals.clear();
             }
+            if lease_to_be_updated {
+                let first_propose_time = committed_entries
+                    .iter()
+                    .find_map(|entry| self.find_propose_time(entry.get_index(), entry.get_term()));
```

Comment:
`find_propose_time` will mutate inner proposal queues, is it expected?

Reply:
Yes, it's expected.

Reply:
So when leader is just transferred, the line will clear all the proposals in queue, hence the commit duration will not be observed?

Reply:
Sorry I don't get it. IMO inside committed entries and proposal queue, entries are sorted with continuous increased index and maybe increased term. And here, `find_propose_time` will 
first process entry with lower index and lower term, if `find_propose_time` can't find `first_propose_time` after process the last entry and hence clear all the proposals in queue, then we can't not find it in any way. If it find `first_propose_time` but clear all the proposals in queue, `last_propose_time` will set to `first_propose_time` because the use of `.or()`

Reply:
/cc @overvenus PTAL